vscode-extension/src/statusBar.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {StatusBarAlignment, StatusBarItem, window} from 'vscode';
+import {RelayExtensionContext} from './context';
+
```

Comment:
@captbaritone good feedback. I think I landed at a healthy middle ground. I put all of the logic relevant to intercepting the `window/showStatus` command & turning those commands into a status bar message into a `LSPStatusBarFeature` class. 

I kept the management of the status bar separate since we might want to use that before the LSP gets booted up.

Left this comment on top of the `LSPStatusBarFeature` class

```
// This StaticFeature is solely responsible for intercepting
// window/showStatus commands from the LSP Server and displaying
// those messages on the client status bar.
//
// The StatusBarItem creation does not happen here since we may
// want to use the status bar to display messages before we
// get messages from the LSP server.
// e.g. Looking for Relay binary...
```